src/Compiler/Microsoft.AspNetCore.Razor.Language/src/DefaultRazorIntermediateNodeLoweringPhase.cs

```diff
@@ -83,9 +83,14 @@ protected override void ExecuteCore(RazorCodeDocument codeDocument)
         var usingReferences = new List<UsingReference>(visitor.Usings);
         for (var j = importedUsings.Count - 1; j >= 0; j--)
         {
-            if (!usingReferences.Contains(importedUsings[j]))
+            var importedUsing = importedUsings[j];
+            if (!usingReferences.Contains(importedUsing) &&
+                // If the using is from the default import, avoid adding it
+                // if a user using exists which is the same except for the `global::` prefix.
+                (!TryRemoveGlobalPrefixFromDefaultUsing(in importedUsing, out var trimmedUsing) ||
+                !usingReferences.Contains(trimmedUsing)))
```

Comment:
Or perhaps just pass a custom comparer to `Contains` to avoid having to have a `UsingReference` type?

Reply:
Though I think we should still avoid the `Substring` call in any case. The list of default usings is small and this will likely need to be done for every single razor file.